# HG changeset patch
# User Rik <rik@octave.org>
# Date 1437091477 25200
#      Thu Jul 16 17:04:37 2015 -0700
# Node ID 63736167fb78075ef9c899a52329611740e77b47
# Parent  b9bd8786d3109b5541bfa1c28dbb6db7bd278459
Fix listdlg() when used without selection mode under Qt (Bug #45545).

* listdlg.m: Use lowercase default for selmode of "multiple".  Use strcmp, not
strcmpi, when checking selmode since all instances are now lower case.

diff --git a/scripts/gui/listdlg.m b/scripts/gui/listdlg.m
--- a/scripts/gui/listdlg.m
+++ b/scripts/gui/listdlg.m
@@ -80,17 +80,17 @@
 
 function [sel, ok] = listdlg (varargin)
 
   if (nargin < 2)
     print_usage ();
   endif
 
   listcell = {""};
-  selmode = "Multiple";
+  selmode = "multiple";
   listsize = [160, 300];
   initialvalue = 1;
   name = "";
   prompt = {};
   okstring = "OK";
   cancelstring = "Cancel";
 
   ## handle key, value pairs
@@ -122,17 +122,17 @@ function [sel, ok] = listdlg (varargin)
   ## make sure listcell strings are a cell array
   if (! iscell (listcell))
     listcell = {listcell};
   elseif (iscellstr (listcell{1}))
     listcell = listcell{1};
   endif
 
   ## make sure valid selection mode
-  if (! strcmpi (selmode, "multiple") && ! strcmpi (selmode, "single"))
+  if (! strcmp (selmode, "multiple") && ! strcmp (selmode, "single"))
     error ("listdlg: invalid SelectionMode");
   endif
 
   if (__octave_link_enabled__ ())
     [sel, ok] = __octave_link_list_dialog__ (listcell, selmode, listsize,
                                              initialvalue, name, prompt,
                                              okstring, cancelstring);
   elseif (__have_feature__ ("JAVA"))
